torch/_inductor/ir.py

```diff
@@ -4680,53 +4705,110 @@ def require_stride1(cls, x):
+                    # size=[s0, 1, 28, 28], stride=[784, 1, 28, 1]), which is not actually necessary.
+
+                    # fix flexiblelayout to be FixedLayout with stride_order
+                    as_storage_and_layout(
+                        x,
+                        freeze=True,
+                        want_contiguous=False,
+                        stride_order=get_stride_order(
+                            V.graph.sizevars.size_hints(x.get_layout().stride)
+                        )
+                        if is_stride_order_storage_and_layout(x, order)
+                        else order,
+                        allow_padding=allow_padding,
+                    )
+                    return x
+                else:
+                    # If the exact_strides is given, freeze the FlexibleLayout to a FixedLayout with the exact_strides.
+                    as_storage_and_layout(
+                        x,
+                        freeze=True,
+                        want_contiguous=False,
+                        stride_order=None,
+                        allow_padding=allow_padding,
+                        exact_strides=exact_strides,
+                    )
+                    return x
+            elif isinstance(x.get_layout(), FixedLayout) and (
+                (order and x.get_layout().is_stride_ordered(order))
+                or (
+                    exact_strides
+                    and (
+                        isinstance(x.data, ReinterpretView)
+                        or list(x.get_layout().stride) == exact_strides
+                    )
+                    and x.get_layout().is_stride_ordered(
+                        get_stride_order(V.graph.sizevars.size_hints(exact_strides))
                     )
-                    if is_stride_order_storage_and_layout(x, order)
-                    else order,
-                    allow_padding=allow_padding,
                 )
-                return x
-            elif isinstance(
-                x.get_layout(), FixedLayout
-            ) and x.get_layout().is_stride_ordered(order):
+            ):
                 return x
             elif isinstance(x.get_layout(), MutationLayoutSHOULDREMOVE):
                 if isinstance(x.get_layout().real_layout(), FlexibleLayout):
                     raise AssertionError(
                         "the MutationLayoutSHOULDREMOVE's real layout shouldn't be FlexibleLayout"
                     )
-                elif isinstance(
-                    x.get_layout().real_layout(), FixedLayout
-                ) and x.get_layout().real_layout().is_stride_ordered(order):
+                elif isinstance(x.get_layout().real_layout(), FixedLayout) and (
+                    (order and x.get_layout().real_layout().is_stride_ordered(order))
+                    or (
+                        exact_strides
+                        and list(x.get_layout().real_layout().stride)
+                        == list(exact_strides)
```

Comment:
Can you factor this out to a function, and ignore strides of dim==1, similar to how we do [here](https://github.com/pytorch/pytorch/blob/44fa9f991c558592917b8c8b1968e5aa9c78304a/torch/_inductor/ir.py#L2924-L2930) ? 

Maybe, `significant_strides_equal` ? (there exists other versions of this, but if we used them, would add guards which we dont want)

Reply:
fixed in https://github.com/pytorch/pytorch/pull/130956/commits/c2b3532ed83271080d57f6393f94742691a27dd5